### PR TITLE
[fix][doc] Avoid refer to deprecated classes

### LINKED ISSUE #8917: Documentation references deprecated MessageBuilder in version 2.7.0.
#### Expected behavior

Example with non deprecated class.

#### Actual behavior

Example uses MessageBuilder class.

#### Steps to reproduce

At: [https://pulsar.apache.org/docs/en/cookbooks-compaction/](https://pulsar.apache.org/docs/en/cookbooks-compaction/)
An import is done through:
**import org.apache.pulsar.client.api.MessageBuilder;**
However, the jar file of 2.7.0 does not contain the interface in Maven central.
As indicated on:
[https://pulsar.apache.org/api/client/org/apache/pulsar/client/api/MessageBuilder.html](https://pulsar.apache.org/api/client/org/apache/pulsar/client/api/MessageBuilder.html)
The interface is deprecated.

#### System configuration
**Pulsar version**: 2.7.0

### PR DESCRIPTION
**Pull request by tisonkun**
This closes #8917.

### Documentation

- [x] `doc` 
- [ ] `doc-required` 
- [ ] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository:

### CODE REVIEW

File: site2/website/versioned_docs/version-2.9.x/cookbooks-compaction.md [hunk 1]
```diff
@@ -109,40 +109,29 @@ Consumer<byte[]> compactedTopicConsumer = client.newConsumer()
...
-        .setContent(someByteArray)
-        .setKey("some-key")
-        .build();
-
-compactedTopicProducer.send(msg);
-
+        compactedTopicProducer.newMessage()
+        .key("some-key")
+        .value(someByteArray)
+        .send();
```
Comment (labuladong) [inline site2/website/versioned_docs/version-2.9.x/cookbooks-compaction.md:136]: Indent seems redundant.

Comment (tisonkun) [reply site2/website/versioned_docs/version-2.9.x/cookbooks-compaction.md]: Good catch! Fixing...

Comment (tisonkun) [reply site2/website/versioned_docs/version-2.9.x/cookbooks-compaction.md]: Fixed.

**Review by labuladong — APPROVED** [review]

**Review by yuruguo — APPROVED** [review]

**Comment by tisonkun** [comment]
Merging...

**Pull request merged**